test/groovy/com/sap/piper/ConfigurationHelperTest.groovy

```diff
@@ -248,4 +248,30 @@ class ConfigurationHelperTest {
     public void testWithMandoryParameterDefaultCustomFailureMessageNotProvidedSucceeds() {
         new ConfigurationHelper([myKey: 'myValue']).withMandatoryProperty('myKey')
     }
+
+    @Test
+    public void testWithMandoryWithFalseCondition() {
+        thrown.none()
+
+        new ConfigurationHelper([enforce: false]).withMandatoryProperty('missingKey', null, {c -> return c.enforce})
+    }
+
+    @Test
+    public void testWithMandoryWithTrueConditionMissingValue() {
+        thrown.expect(IllegalArgumentException)
+        thrown.expectMessage('ERROR - NO VALUE AVAILABLE FOR missingKey')
+        
+        new ConfigurationHelper([enforce: true]).withMandatoryProperty('missingKey', null, {c -> return c.execute})
```

Comment:
Comparing the testName and the map ```[enforce:true]``` with ```return c.execute``` I believe there is a mismatch. Shouln't it be ```return c.enforce```?